File: StemExplorerAPI/StemExplorerAPI/Models/Entities/Challenge.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace StemExplorerAPI.Models.Entities
{
    public class Challenge
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public Enums.ChallengeCategories Category { get; set; }
        
        // EF relationship definition
        public int LocationId { get; set; }
        public Location Location { get; set; }

        public List<ChallengeLevel> ChallengeLevels { get; set; }
    }
}

File: stem-explorer-ng/src/challenge/models/challenge.ts
import { Levels } from 'src/app/shared/enums/levels.enum';
import { AnswerType } from 'src/app/shared/enums/answer-type.enum';
import { SafeResourceUrl } from '@angular/platform-browser';

export interface Challenge {
    id: number;
    title: string;
    description: string;
    category: number;
    locationId: number;
    challengeLevels: ChallengeLevel[];
}

export interface ChallengeLevel {
    uid: number;
    questionText: string;
    questionImage?: string;
    questionImageHelperText?: string;
    instructions: string;
    instructionsImage?: string;
    instructionsImageHelperText?: string;
    difficulty: Levels;
    hint: string;
    answerType: AnswerType;
    answer: string[];
    possibleAnswers?: string[];
    complete?: boolean;
    videoEmbedUrl?: SafeResourceUrl;
}
